src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -455,135 +457,163 @@ private void SetNestedLanguageBlockASTGuids(Guid guid, List<ProtoCore.AST.Node>
             }
         }
 
-
-        private IEnumerable<AssociativeNode> GetDeltaAstListModified(IEnumerable<Subtree> modifiedSubTrees)
+        private void UpdateCachedASTFromSubtrees(List<Subtree> modifiedSubTrees)
         {
-            var deltaAstList = new List<AssociativeNode>();
-            csData.RemovedBinaryNodesFromModification = new List<AssociativeNode>();
-            csData.RemovedFunctionDefNodesFromModification = new List<AssociativeNode>();
-            csData.ModifiedFunctions = new List<AssociativeNode>();
-            csData.ForceExecuteASTList = new List<AssociativeNode>();
-            csData.ModifiedNestedLangBlock = new List<AssociativeNode>();
-
-            if (modifiedSubTrees == null)
+            if (modifiedSubTrees != null)
             {
-                return deltaAstList;
+                for (int n = 0; n < modifiedSubTrees.Count(); ++n)
+                {
+                    Subtree subtree = modifiedSubTrees[n];
+                    if (subtree.AstNodes == null)
+                    {
+                        continue;
+                    }
+
+                    UpdateCachedASTList(subtree, subtree.ModifiedAstNodes);
```

Comment:
Yes, there are instances where the UI passes a null subtree. This is code that was moved from GetDeltaAstListModified